export.go

```diff
@@ -20,35 +20,22 @@ import (
 	"context"
 	"io"
 
-	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/images/oci"
+
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
-type exportOpts struct {
```

Comment:
I think it would be more consistent to just add the all platforms option here, then it can use that to create the V1 exporter. I kind of think we need to do some deeper cleaner, so we can focus on the refactoring and interfaces in one change and just focus on the platform feature and bug fix here.

Reply:
@dmcgowan   i have changed `exportOpts` struct to  `type V1ExporterOpt func(c *V1Exporter) error` .  and use `ResolveV1ExportOpt ` function to create the `V1Exporter` .    if i  add `AllPlatforms ` feild to `exportOpts` struct, i  must use `exportOpts` to create `V1Exporter`.  Suppose if we need to add new fields to `V1Exporter` struct  in the future, we also need add same field to  `exportOpts` struct.   Is this more cumbersome?